test_wrapper/test_wrapper.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"time"
+
+	"cloud.google.com/go/storage"
+	junitFormatter "github.com/jstemmer/go-junit-report/formatter"
+	junitParser "github.com/jstemmer/go-junit-report/parser"
+)
+
+const (
+	metadataURLPrefix   = "http://metadata.google.internal/computeMetadata/v1/instance/attributes/"
+	testTxtResult       = "go-test.txt"
+	testResultObject    = "outs/junit_go-test.xml"
+	testBinaryLocalPath = "image_test"
+	artifactPath        = "/artifact"
+)
+
+func main() {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatalf("failed to create cloud storage client: %v", err)
+	}
+
+	testBinaryURL, err := getMetadataAttribute("_test_binary_url")
+	if err != nil {
+		log.Fatalf("failed to get metadata _test_binary_url: %v", err)
+	}
+
+	u, err := url.Parse(testBinaryURL)
```

Comment:
no need to parse here, parse in downloadGCSObject

Reply:
Move to downloadGCSObject